test/specs/reference/tests/02.related_reference.js

```diff
@@ -470,6 +470,61 @@ exports.execute = function(options) {
                     var ref = pageWithToName.tuples[0].getAssociationRef({"not the id":9003});
                     expect(ref).toBe(null);
                 });
+
+                describe('Tuple.getBatchAssociationRef, ', function () {
+                    var batchPage, selectedTuples;
+                    it('should return an array of references including the filtered assocation reference.', function() {
+                        var url = options.url + "/catalog/" + catalog_id + "/entity/";
+                        // there are 3 rows in the set for "pageWithToName"
+                        related[2].sort([{"column":"id", "descending":false}]).read(3).then(function(response) {
+                            batchPage = response;
+                            expect(batchPage.tuples.length).toBe(3, "pageWithToName set is not of size 3");
+
+                            selectedTuples = [batchPage.tuples[0], batchPage.tuples[1]];
+                            var batchRefs = batchPage.tuples[0].reference.getBatchAssociationRef(selectedTuples);
+                            expect(batchRefs.length).toBe(1, "more than 1 reference was returned");
+                            expect(batchRefs[0].uri).toEqual(url + "reference_schema:reference_table/id=9003/(id)=(reference_schema:association_table_with_toname:id_from_ref_table)/(id_from_inbound_related_table=1);(id_from_inbound_related_table=2)", "batch reference uri is incorrect");
+
+                            return batchRefs[0].read(5);
+                        }).then(function(batchResponse) {
+                            expect(batchResponse.data.length).toBe(2);
+
+                            done();
+                        }, function(err) {
+                            console.dir(err);
+                            done.fail();
+                        });
+                    });
+
+                    it('should return 2 references if the number of tuples to create an association for exceeds the url length limit.', function () {
+                        related[2].sort([{"column":"id", "descending":false}]).read(3).then(function(response) {
+                            var batchPage = response;
+
+                            // modify length limit to make testing this easier
+                            var oldLimit = options.ermRest.URL_PATH_LENGTH_LIMIT;
+                            options.ermRest.URL_PATH_LENGTH_LIMIT = 500;
```

Comment:
I'm not sure how I feel about this. I understand you didn't want to add data to those two tables (the association and leaf table), but this doesn't feel right at all. You don't need to add too many rows. Just make sure the values that you're using are long. The name of the columns are already long too, so that should help. As I mentioned [I did the same](https://github.com/informatics-isi-edu/ermrestjs/blob/master/test/specs/column/conf/pseudo_column_schema/data/inbound%204%20long%20table%20name.json) for pseudo-column aggregate test cases.

Reply:
Won't take long to add lengthy data, was trying to cut a corner.